app/db/index.js

```diff
@@ -74,23 +87,34 @@ function addNewMarket(info) {
  * @param {*} info All question answers from the assessment page.
  */
 function updateExistingMarket(info) {
-  const marketsRef = db.child(info.marketInfo.marketName);
-
-  // Update market level.
-  marketsRef.child('marketInfo').update({
-    marketLevel: info.level,
-  });
+    // Get current values to avoid overwriting current values.
+    getSpecificMarket(info.marketInfo.marketName).then(market => {
+        const update = {
+            $set: {
```

Comment:
make `update` the value of `$set` so that `update` is just the document itself and not the MongoDB client's update command, and then do `{ $set: update }` when you're actually calling the update command